VDA5050_EN.md

```diff
@@ -903,7 +933,21 @@ An exception to this rule is, if the AGV has to pause on the edge (because of a
 ![Figure 13 nodeStates, edgeStates, actionStates during order handling](./assets/Figure13.png)
 >Figure 13 nodeStates, edgeStates, actionStates during order handling
 
+#### Corridors
+
+The corridor attribute of an edge leads to two different types of nodes inside inside an order: *goal nodes* and *way nodes*.
+
+- A *goal node* contains actions and therefore the shuttle has to reach this nodes precisely. The fleet management can define via the `allowedDeviationXY` and `allowedDeviationTheta` how precisely the node has to be reached. If no deviation is defined no deviation is allowed (no deviation means within the normal tolerance of the AGV manufacturer).
+- A *way node* contains no actions and therefore the shuttle may pass this node not precisely. The node attributes  `allowedDeviationXY` and `allowedDeviationTheta`  have no effect.
+*(Remarks: Consulting  `allowedDeviationXY` and `allowedDeviationTheta` to detect whether a way point is passed or not makes it difficult to define a correct ridable order and not defeating the reason using the corridor attribute. While the corridor attribute covers the whole vehicle contour an allowed deviation refers to the reference point of the vehicle. This makes it not trivial to define right deviation range according to the given corridor polygon or vice versa. Way points should be used defining the navigation area together with the corridor attribute and traffic control.)*
+
+An AGV on an edge with a corridor attribute shall determine the current edge by calculating the perpendicular distance to each edge of the current base. The edge with the shortest distance is the current edge. All nodes and edges with lower sequence numbers count as passed.
```

Comment:
Therefore the polygons shall overlap so that the vehicle is inside the polygon after an "edge change". In my first proposal I leave this point open how a vehicle determines when a node is passed (like the current text in chapter 6.10.2 for AGVs: "The AGV decides on its own, when a node should count as traversed."). The majority of the core group member asks for a definition.
Overall, I think this definition goes along with chapter 6.10.2 and if an edge is not longer part of the base the corridor polygon of the edge should also not be part of the base any longer.

Reply:
I think this might create problems in some cases. Imagine an edge leading though an narrow environment where the agv/amr can not turn (say from node A to B). Lets define this e(A,B). Later e(B,A) is released for that vehicle as well while traversing over e(A,B). The rule enforces an updates that invalidates all released nodes between e(A,B) -> ..... e(B,A). The agv/amr is stuck on edge e(B,A) with the wrong orientation and is unable to execute the order anymore.

Reply:
The definition of the "perpendicular distance" is also a bit troublesome. I suggest to clarify that requirement, too. The edge is not a line but a line segment, so the end points of that line must be taken into account. Otherwise an edge that is not even a part of the currently traversed corridor will force an update that invalidates the order.

Reply:
Yes, interesting point, maybe I find a definition which avoids this problem. Remark: also the current standard can lead  into problematic situation (e.g choosing wrong deviation ranges).

Reply:
I suggest something like this:
*(a) an edge must be counted as passed as soon as the agv/amr detects that it left the zone/corridor that is associated with that edge
*(b) It is required that an increase of the lastNodeID does not violate that the shape of the vehicle is in the remaining navigation space (specification needed waynode, goalnode...)
*(c) The vendor specific implementation might increase the lastNodeId sooner as long as they do not violate (b)

This gives the vendor the opportunity to increase the lastNodeID if they think it is possible but it will enforce that an update happens as soon as the space is not needed anymore.

Reply:
I have a longer internal discussion and I will try to stick with the (original) idea not to specify it. I will try to find a similar sentence analog to

> The AGV decides on its own, when a node should count as traversed. Generally, the AGV’s control point should be within the node’s deviationRangeXY and its orientation within deviationRangeTheta .

Of cause we have to solution for our vehicle and our use-case, but don't claim this is a good solution for everyone and have to be specified.